lib/view_component/engine.rb

```diff
@@ -30,7 +31,16 @@ class Engine < Rails::Engine # :nodoc:
       end
 
       ActiveSupport.on_load(:view_component) do
-        options.each { |k, v| send("#{k}=", v) }
+        virtual_options = [:instrumentation_enabled]
+        options.select { |k, _| virtual_options.exclude?(k) }.each { |k, v| send("#{k}=", v) }
```

Comment:
I read this a few times and found it difficult to follow. Is there a way to make what this is trying to do more explicit, or is it possibly unnecessary?

Reply:
Checked with `respond_to?("#{k}=")`, an explicit variant is a bit verbose for my taste.